web/libs/storybook/addons/theme-toggle/theme-toggle.tsx

```diff
@@ -0,0 +1,35 @@
+import React from "react";
+
+import { useStorybookApi } from "@storybook/manager-api";
+import { IconButton } from "@storybook/components";
+import { MoonIcon, SunIcon } from "@storybook/icons";
+
+import { ADDON_ID, TOOL_ID, THEMES, DEFAULT_THEME } from "./constants";
+import { useAtom, useAtomValue } from "jotai/react";
+import { evaluatedThemeAtom, themeAtom } from "./atoms";
+
+export const ThemeTool = React.memo(function MyAddonSelector() {
+  const [theme, setTheme] = useAtom(themeAtom);
+  const evaluatedTheme = useAtomValue(evaluatedThemeAtom);
+  const api = useStorybookApi();
+
+  const toggleTheme = React.useCallback(() => {
+    setTheme((previousTheme) => THEMES[(THEMES.indexOf(previousTheme) + 1) % THEMES.length]);
+  }, []);
+
+  React.useEffect(() => {
```

Comment:
```suggestion
  useEffect(() => {
```

for the sake of consistency